src/mqtt/protocol/mqtt/mqtt_client.c

```diff
@@ -450,15 +457,19 @@ mqtt_send_msg(nni_aio *aio, mqtt_ctx_t *arg)
 		packet_id = nni_mqtt_msg_get_packet_id(msg);
 		taio = nni_id_get(&p->sent_unack, packet_id);
 		if (taio != NULL) {
-			nni_plat_printf("Warning : msg %d lost due to "
-			                "packetID duplicated!",
-			    packet_id);
+			log_warn("msg %d lost due to packetID duplicated!", packet_id);
 			nni_aio_finish_error(taio, NNG_ECANCELED);
+			if ((tmsg = nni_aio_get_msg(taio)) != NULL)
```

Comment:
possibility to clash with cb of taio. leave msg to the cb(nng_mqtt_client_send_cb)